### DISCUSSION TITLE
full node_modules for single package

### DISCUSSION THREAD

**Discussion by mokone91** [pnpm/pnpm#5326] [General]
How can i get node_modules folder with all packaged related to current package in workspace?
its needed to build serverless lambda layer for nestjs app.

Currently i cpy package.json to separate folder and run npm i inside...

Thanks!

**Comment by zkochan** [comment]
Try `pnpm deploy` 

https://pnpm.io/cli/deploy

**Reply by mokone91** [reply]
Before:

```
#cp ${CURRENT_DIR}/../package.json layers/modules/nodejs/package.json
#cd layers/modules/nodejs
#npm install --omit=dev --force --no-package-lock
#cd ${CURRENT_DIR}/../
```

After: 
```
pnpm --filter=api deploy --prod layers/modules/nodejs
```
In second case i get projects sources in layer - its not a problem, but after serverless will pack layers/modules - unzipped folder size is 350mb,  before was 197 mb.. labda layer limit is 262mb

**Reply by zkochan** [reply]
What about if you use node-linker hoisted?

**Reply by mokone91** [reply]
in this case there is only 'layers/modules/nodejs/node_modules/.pnpm/lock.yaml' created
i also use shamefully-hoist=true, ui packages in monorepo use storybook, and its cant work without this flag

**Reply by zkochan** [reply]
are there no packages in `layers/modules/nodejs/node_modules/`? Then it is probably a bug. `pnpm deploy` is a pretty new command maybe it doesn't work with node-linker=hoisted. but it should. It should be fixed.

**Reply by bm77525-kr** [reply]
Any update on this? My `node_modules` also ends up empty when using the `deploy` command with `node-linker=hoisted` (symlinks don't work with azure functions).

**Reply by menseAjit** [reply]
any update on this issue?  i am also having issue packaging lambda.

**Reply by mokone91** [reply]
seems no( here is my tmp workaround https://github.com/serverless-heaven/serverless-webpack/issues/422#issuecomment-1318753747
